### PR TITLE
[1/2] Deployment Preparation: ERC-4626 Linear Pool Factory V4

### PR DESCRIPTION
**Pull request by joaobrunoah**
# Deployment preparation: `ERC4626LinearPoolFactory`
Build-info (linear-pools repo): [9780842b65e6489aa4db6002f159719500c3a64c](https://github.com/orbcollective/linear-pools/commit/9780842b65e6489aa4db6002f159719500c3a64c)

# Deployment preparation tasks checklist:

- [x] Deployments README is updated with the new task name and link
- [x] Task README has a description and links to the artifacts and expected output folders
- [x] build-info is updated with artifacts compiled at the specified commit
- [x] Artifacts are generated from build-info
- [x] New task has a working fork test

# Deprecating old tasks checklist:

- [x] Old task was moved to deprecated/ directory (already done)
- [x] Deployments README is updated: old task is moved to the 'deprecated' section, and links are updated with the deprecated/ prefix
- [x] Old task imports in index.ts, input.ts and fork tests are updated
- [x] Old task README is updated with a warning sign, a link to the replacement task and a short description

# Code checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented in tests and scripts
- [x] Fork test block number and test cases are appropriate for this deployment
- [x] The base branch is either master, or there's a description of how to merge

# Issue Resolution
N/A

### CODE REVIEW

File: pkg/deployments/README.md [hunk 1]
```diff
@@ -97,7 +97,6 @@ Returns an object with all contracts from a deployment and their addresses.
 | L2 Gauge Checkpointer                                  | [`20221205-l2-gauge-checkpointer`](./tasks/20221205-l2-gauge-checkpointer)                           |
 | VeBoost V2                                             | [`20221205-veboost-v2`](./tasks/20221205-veboost-v2)                                                 |
 | Gauge Registrant V3, supporting the Adaptor entrypoint | [`20230109-gauge-adder-v3`](./tasks/20230109-gauge-adder-v3)                                         |
-| Linear Pools for ERC4626 Tokens V3                     | [`20230206-erc4626-linear-pool-v3`](./tasks/20230206-erc4626-linear-pool-v3)                         |
```
Comment (EndymionJkb) [inline pkg/deployments/README.md:100]: Needs active entry for V4

File: pkg/deployments/tasks/20230409-erc4626-linear-pool-v4/test/contracts/ReadOnlyReentrancyAttackerLPV4.sol [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import "@balancer-labs/v2-interfaces/contracts/pool-linear/ILinearPool.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/ERC20Helpers.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+
+/**
+ * @notice Performs a read-only reentrancy attack on a target linear pool, making use of the `receive` callback hook
+ * in the middle of an exit operation.
+ */
+contract ReadOnlyReentrancyAttackerLPV4 {
```
Comment (EndymionJkb) [inline pkg/deployments/tasks/20230409-erc4626-linear-pool-v4/test/contracts/ReadOnlyReentrancyAttackerLPV4.sol:28]: Is this different from ReadOnlyReentrancyAttackerLP? Can we just reuse the old one like with the others?

File: pkg/deployments/tasks/20230409-erc4626-linear-pool-v4/test/contracts/MockERC4626TokenV4.sol [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-interfaces/contracts/pool-linear/IERC4626.sol";
+import "@balancer-labs/v2-pool-utils/contracts/test/MaliciousQueryReverter.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/test/TestToken.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+
+contract MockERC4626TokenV4 is TestToken, IERC4626, MaliciousQueryReverter {
```
Comment (EndymionJkb) [inline pkg/deployments/tasks/20230409-erc4626-linear-pool-v4/test/contracts/MockERC4626TokenV4.sol:24]: Is this different from MockERC4626Token? The others reused the mocks.

File: pkg/deployments/tasks/deprecated/20230206-erc4626-linear-pool-v3/output/arbitrum.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
-{
-  "ERC4626LinearPoolFactory": "0xa3B9515A9c557455BC53F7a535A85219b59e8B2E",
-  "MockERC4626Token": "0x62aaB12865d7281048c337D53a4dde9d770321E6",
-  "MockERC4626LinearPool": "0xf984eb2B8a7eF780245a797A2fCcd82f346409ca"
+{
```
Comment (EndymionJkb) [inline pkg/deployments/tasks/deprecated/20230206-erc4626-linear-pool-v3/output/arbitrum.json:1]: Why is there a diff in any of these outputs? Formatting issue? Should just be moved without changes.

**Review by EndymionJkb — COMMENTED** [review]
Build info matches. See comments

File: pkg/deployments/README.md (hunk 1 above)
Comment (joaobrunoah) [reply pkg/deployments/README.md:100]: Done!

File: pkg/deployments/tasks/20230409-erc4626-linear-pool-v4/test/contracts/ReadOnlyReentrancyAttackerLPV4.sol (hunk 2 above)
Comment (joaobrunoah) [reply pkg/deployments/tasks/20230409-erc4626-linear-pool-v4/test/contracts/ReadOnlyReentrancyAttackerLPV4.sol]: Removed it!

File: pkg/deployments/tasks/20230409-erc4626-linear-pool-v4/test/contracts/MockERC4626TokenV4.sol (hunk 3 above)
Comment (joaobrunoah) [reply pkg/deployments/tasks/20230409-erc4626-linear-pool-v4/test/contracts/MockERC4626TokenV4.sol]: Removed it!

File: pkg/deployments/tasks/deprecated/20230206-erc4626-linear-pool-v3/output/arbitrum.json (hunk 4 above)
Comment (joaobrunoah) [reply pkg/deployments/tasks/deprecated/20230206-erc4626-linear-pool-v3/output/arbitrum.json:1]: I think it's a git thing, but there was no reformatting. The files were marked as deleted in one folder and as added in another, and I'm not being able to change it. Still, I just copied the original files, pasted them in the folder, and committed, to make sure no formatting happened, and these files are still being marked as changed

**Review by EndymionJkb — APPROVED** [review]

**Pull request merged**